File: FunASR/examples/industrial_data_pretraining/bicif_paraformer/tokenizer.py
import json


class Tokenizer:
    def __init__(self, vocab_path=""):
        with open(vocab_path, "r", encoding="utf-8") as f:
            vocab = json.load(f)
        self.vocab = vocab

    def id2token(self, ids):
        tokens = [self.vocab[id] for id in ids]

        return tokens

File: mainapp/lib/tokenizer.dart
import 'dart:convert';

import 'package:flutter/services.dart';

class Tokenizer {
  late List<dynamic> vocab;

  Future<void> init() async {
    vocab = await loadJsonFromAssets();
  }

  Future<List<dynamic>> loadJsonFromAssets() async {
    String jsonString = await rootBundle.loadString("assets/tokens.json");
    return jsonDecode(jsonString);
  }

  List<String> id2Token(List<int> id){
    List<String> token = List.empty(growable: true);
    for (var e in id) {
      token.add(vocab[e]);
    }
    return token;
  }

}
